src/librustc/infer/fudge.rs

```diff
@@ -141,12 +167,10 @@ impl<'a, 'gcx, 'tcx> TypeFolder<'gcx, 'tcx> for RegionFudger<'a, 'gcx, 'tcx> {
 
     fn fold_region(&mut self, r: ty::Region<'tcx>) -> ty::Region<'tcx> {
         match *r {
-            ty::ReVar(v) if self.region_vars.contains(&v) => {
+            ty::ReVar(vid) if self.region_vars.contains(&vid) => {
                 self.infcx.next_region_var(self.origin.clone())
```

Comment:
Why doesn't this reuse the origin from the original variable, just like for types?

cc @nikomatsakis

Reply:
It does now.